Userland/Utilities/mount.cpp

```diff
@@ -217,19 +218,60 @@ static ErrorOr<void> mount_using_loop_device(int inode_fd, StringView mountpoint
     return result;
 }
 
+static ErrorOr<void> apply_file_system_specific_options(int mount_fd, Vector<String> const& filesystem_specific_options)
+{
+    for (auto& option : filesystem_specific_options) {
+        auto key_value_option = TRY(option.split_limit('=', 2));
+        MountSpecificFlag flag;
+        flag.key_string_length = key_value_option[0].bytes().size();
+        flag.key_string_addr = key_value_option[0].bytes().data();
+        if (key_value_option.size() > 1 && !key_value_option[1].is_empty()) {
+            auto possible_int_value = key_value_option[1].to_number<i64>();
+            auto possible_uint_value = key_value_option[1].to_number<u64>();
+            if (possible_uint_value.has_value()) {
+                flag.value_type = MountSpecificFlag::ValueType::UnsignedInteger;
+                flag.value_length = 8;
+                u64 value = possible_uint_value.value();
+                flag.value_addr = &value;
+                TRY(Core::System::ioctl(mount_fd, MOUNT_IOCTL_SET_MOUNT_SPECIFIC_FLAG, &flag));
+            } else if (possible_int_value.has_value()) {
+                VERIFY(!possible_uint_value.has_value());
+                flag.value_type = MountSpecificFlag::ValueType::SignedInteger;
+                flag.value_length = 8;
+                u64 value = possible_int_value.value();
+                flag.value_addr = &value;
+                TRY(Core::System::ioctl(mount_fd, MOUNT_IOCTL_SET_MOUNT_SPECIFIC_FLAG, &flag));
+            } else {
+                flag.value_type = MountSpecificFlag::ValueType::ASCIIString;
+                flag.value_addr = key_value_option[1].bytes().data();
+                flag.value_length = key_value_option[1].bytes().size();
+                TRY(Core::System::ioctl(mount_fd, MOUNT_IOCTL_SET_MOUNT_SPECIFIC_FLAG, &flag));
+            }
+            continue;
+        }
+        flag.value_type = MountSpecificFlag::ValueType::Boolean;
+        flag.value_addr = nullptr;
+        flag.value_length = 0;
+        TRY(Core::System::ioctl(mount_fd, MOUNT_IOCTL_SET_MOUNT_SPECIFIC_FLAG, &flag));
+    }
+    return {};
+}
+
 ErrorOr<int> serenity_main(Main::Arguments arguments)
 {
     StringView source;
     StringView mountpoint;
     StringView fs_type;
     StringView options;
+    Vector<String> filesystem_specific_options;
     bool should_mount_all = false;
 
     Core::ArgsParser args_parser;
     args_parser.add_positional_argument(source, "Source path", "source", Core::ArgsParser::Required::No);
     args_parser.add_positional_argument(mountpoint, "Mount point", "mountpoint", Core::ArgsParser::Required::No);
     args_parser.add_option(fs_type, "File system type", nullptr, 't', "fstype");
     args_parser.add_option(options, "Mount options", nullptr, 'o', "options");
+    args_parser.add_option(filesystem_specific_options, "Mount filesystem-specific options", "fs-specific-option", 'O', "filesystem-specific options");
```

Comment:
I had to read this description thrice to actually parse it. Can it just say “filesystem-specific options” instead?